src/gui/src/SongEditor/SongEditorPanel.cpp

```diff
@@ -550,6 +550,7 @@ void SongEditorPanel::updatePlaybackTrackIfNecessary()
 void SongEditorPanel::updatePositionRuler()
 {
 	m_pPositionRuler->createBackground();
+	m_pPositionRuler->update();
```

Comment:
It looks like the only call site for `updatePositionRuler` is `updateAll`, which has a call to `updateTimelineUsage` just after this, which *also* calls createBackground. So I think the ruler is needlessly repainted (to its background pixmap) twice. 

Also the other component widgets are explicitly `->update()`ed in updateAll, so it would make sense to do that there rather than here?

Reply:
> It looks like the only call site for `updatePositionRuler` is `updateAll`, which has a call to `updateTimelineUsage` just after this, which _also_ calls createBackground. So I think the ruler is needlessly repainted (to its background pixmap) twice.

I should clarify -- that's a comment on the existing code, not yours :)